CONTRIBUTING.md

```diff
@@ -0,0 +1,133 @@
+
+## Feature requests
+
+Feature requests are welcome, but take a moment to find out whether your idea fits with the scope and aims of the project. It's up to *you* to make a strong case to convince the project's developers of the merits of this feature. Please provide as much detail and context as possible.
+
+# Engineering standards
+
+All ongoing development for an upcoming release gets commited to the **`main`** branch. The `main` branch technically serves as the "development" branch as well, but all code that is committed to the `main` branch should be considered *stable*.
+
+Ensure any changes are clear and well-documented. When we say "well-documented":
+
+* If the changes include code, ensure all additional code has documentation in and around it. This includes documenting the definitions of fnuctoins, statements in code, sections.
+* The most helpful code comments explain why, establish context, or efficiency summarize how. Avoid simply repeating details from declarations. When in doubt, favour overexplaining to underexplaining.
+* Code comments should be consistent with their languge conventions. For example, please use [JSDoc](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html) for TypeScript/JavaScript, or [NatSpec](https://docs.soliditylang.org/en/v0.8.13/natspec-format.html) for Solidity.
+* Any new features must have corresponding user documentation. Any removed features must have their user documentation removed from the documents.
+* Do **NOT** submit commented-out code. If the code does not need to be used anymore, please remove it.
+* While `TODO` comments are frowned upon, every now and then it is OK to put a `TODO` to note that a particular section of code needs to be worked on in the future. However, it is also known that "TODOs" often do not get worked on, and as such, it is more likely you will be asked to complete the TODO at the time you submit it.
+* Write clear, descriptive [commit messages](#commit-messages).
+
+Please provide unit tests with your code if possible. If unable to provide a unit test, please provide an explanation as to why in your pull request, including a description of the steps used to manually verify the changes.
+
+Ensure your commits are atomic. Each commit tells a story about what changes are being made. This makes it easier to identify when a bug is introduced into the codebase, and as such makes it easier to fix.
+
+All commits must either be rebased in atomic order or squashed (if the squashed commit is considered atomic). Merge commits are not accepted. All conflicts must be resolved prior to pushing changes.
+
+**All pull requests should be made from the `main` branch**. All PRs to `main` branch shall be subject to review. Repository restrictions shall be setup to prohibit the merging of code that has not been reviewed.
```

Comment:
What is the reason for this requirement? Locally, when a developer works on a feature he creates a feature-related branch. I guess, a related PR have to be created from this feature-related branch.
Maybe "to the `main` branch" you mean?

Reply:
Yep, sure, I think it was a brain fart of mine and I got mixed up on "to" and "from" and that perspective kind of stuff. Will adjust :+1: